Templates/CSharp/Model/EntityType.cs.tt

```diff
@@ -75,7 +75,19 @@ namespace <#=entity.Namespace.GetNamespaceName()#>
         }
     <#
         }
+		else
+		{
+	#>
 
+		///<summary>
+		/// The <#=entityName#> constructor
+		///</summary>
+        public <#=cstorTypeDeclaration#>()
+        {
+            this.ODataType = "<#=entity.FullName#>";
```

Comment:
What would happen if developers put something else in here?

Should the `string ODataType { get; set; }` property have no setter?
`string ODataType { get; }`

Reply:
The service may not properly deserialize the request if a developer sets the Odata.type property to a value that matches the object. We do need at least the getter and serialization instructions. This still needs work.

Reply:
Nevermind the last comment. The complex type template has the getter/setter.